src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -2968,31 +2970,33 @@ class spell_pri_spirit_of_redemption : public AuraScript
     }
 };
 
-// 314867 - Shadow Covenant
+// 8092 - Mind Blast
 class spell_pri_shadow_covenant : public SpellScript
 {
-    bool Validate(SpellInfo const* spellInfo) override
+    bool Validate(SpellInfo const* /*spellInfo*/) override
     {
-        return ValidateSpellEffect({ { spellInfo->Id, EFFECT_2 } });
+        return ValidateSpellInfo({ SPELL_PRIEST_SHADOW_COVENANT, SPELL_PRIEST_SHADOW_COVENANT_EFFECT });
     }
 
-    void FilterTargets(std::list<WorldObject*>& targets) const
+    bool Load() override
     {
-        // remove explicit target (will be readded later)
-        Trinity::Containers::Lists::RemoveUnique(targets, GetExplTargetWorldObject());
-
-        // we must remove one since explicit target is always added.
-        uint32 maxTargets = uint32(GetEffectInfo(EFFECT_2).CalcValue(GetCaster()) - 1);
+        if (!GetCaster()->HasAura(SPELL_PRIEST_SHADOW_COVENANT))
+            return false;
 
-        Trinity::SelectRandomInjuredTargets(targets, maxTargets, true);
+        return true;
+    }
 
-        if (Unit* explicitTarget = GetExplTargetUnit())
-            targets.push_front(explicitTarget);
+    void TriggerShadowCovenant() const
```

Comment:
Would suggest renaming it in a way so it reflects the used hook, like: `HandleAfterCast`